### PR TITLE
Fix static analysis warnings

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
The PR fixes static analysis warnings in the `stages` package.

### CODE REVIEW

File: zk/stages/stage_dataStreamCatchup.go [hunk 1]
```diff
@@ -99,7 +102,7 @@ func CatchupDatastream(ctx context.Context, logPrefix string, tx kv.RwTx, stream
 			// without execution we save progress block by block and can populate an unfinished batch to stream
 			// otherwise we get a batch gap in the stream because a few blocks don't get written to the stream
 			// and then execution continues from the middle of the batch
-			latestBatch, err = stages.GetStageProgress(tx, stages.HighestSeenBatchNumber)
+			_, err = stages.GetStageProgress(tx, stages.HighestSeenBatchNumber)
```
Comment (hexoscott) [inline zk/stages/stage_dataStreamCatchup.go]: I'm not sure I get this change, do we not want to set the variable?

**Review by hexoscott — COMMENTED** [review]
One little thing I'm not sure about

File: zk/stages/stage_dataStreamCatchup.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply zk/stages/stage_dataStreamCatchup.go]: The funny part is that the `latestBatch` is not used in this else branch at all, I'm wondering if we should remove this part altogether https://github.com/0xPolygonHermez/cdk-erigon/pull/927/files#diff-8f84bd2ac0843ef0e0e87aab4cfb56bcc065cd9e4b577982d6009145b06de0a6R105-R108? It seems to me like it is not necessary at all.

```go
_, err = stages.GetStageProgress(tx, stages.HighestSeenBatchNumber)
if err != nil {
    return 0, err
}
```

Comment (hexoscott) [reply zk/stages/stage_dataStreamCatchup.go]: Ah I see now in the wider scope, yep, makes sense to just keep that local to the `hasExecutors` branch

Comment (Stefan-Ethernal) [reply zk/stages/stage_dataStreamCatchup.go]: Removed the redundant call https://github.com/0xPolygonHermez/cdk-erigon/pull/927/commits/86126132592b7ccde12d9a49573a4766435bc415

**Comment by Stefan-Ethernal** [comment]
@hexoscott this is also interesting snippet, as this slice seems not to be used: https://github.com/0xPolygonHermez/cdk-erigon/pull/927/files#diff-a9e687340326b210454393f878ea9bd636eb505d4b1b08598c61f625f6b90571R135, although if we remove the slice, there is some more logic that can be removed.

**Comment by hexoscott** [comment]
> @hexoscott this is also interesting snippet, as this slice seems not to be used: https://github.com/0xPolygonHermez/cdk-erigon/pull/927/files#diff-a9e687340326b210454393f878ea9bd636eb505d4b1b08598c61f625f6b90571R135, although if we remove the slice, there is some more logic that can be removed.

I'd say we can remove that.  I think this is a hangover from where we split out the l1 info tree handling into its own stage completely.

**Review by hexoscott — DISMISSED** [review]

**Comment by hexoscott** [comment]
Looks good but looks like we need the latest code merging in to handle conflicts

**Comment by Stefan-Ethernal** [comment]
> Looks good but looks like we need the latest code merging in to handle conflicts

@hexoscott Conflicts are resolved, seems like we need to be fast on a trigger when merging the PR, because of rate of changes on the `zkevm` branch :sweat_smile:

**Review by hexoscott — APPROVED** [review]

**Comment by hexoscott** [comment]
just timing on this one, we just merged a large PR

**Pull request merged**